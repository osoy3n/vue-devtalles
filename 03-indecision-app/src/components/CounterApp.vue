<script lang="ts" setup>
import { useCounter } from '@/composables/useCounter'

// interface Props {
//   value: number
// }

// const props = defineProps<Props>()

const { counter, squareCounter } = useCounter()

// import { computed, ref } from 'vue'

// Option with JS
// const props = defineProps({
//   value: { type: Number, required: true },
// })

// const counter = ref(props.value)
// const squareCounter = computed(() => counter.value * counter.value)
</script>

<template>
  <section>
    <h3>Counter: {{ counter }}</h3>
    <h3>Square: {{ squareCounter }}</h3>

    <div>
      <button class="btn" @click="counter++">+1</button>
      <button class="btn" @click="counter--">-1</button>
    </div>
  </section>
</template>

<style scoped>
@reference "tailwindcss";
.btn {
  @apply p-2 bg-blue-500 rounded hover:bg-blue-700 mr-2;
}
</style>
