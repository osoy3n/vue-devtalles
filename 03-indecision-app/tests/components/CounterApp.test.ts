import { describe, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'
import CounterApp from '@/components/CounterApp.vue'

describe('<CounterApp />', () => {
  test('should match snapshot', () => {
    const wrapper = mount(CounterApp, {
      props: {
        value: 5,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('render the counter value correctly', () => {
    const value = 10
    const wrapper = mount(CounterApp, {
      props: {
        value: value,
      },
    })

    expect(wrapper.find('h3').text()).toContain(`Counter: ${value}`)
    expect(wrapper.find('[data-testid="square-label"]').text()).toContain(
      `Square: ${value * value}`,
    )
  })
})
