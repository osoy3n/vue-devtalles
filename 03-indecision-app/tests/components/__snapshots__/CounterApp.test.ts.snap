// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<CounterApp /> > should match snapshot 1`] = `
"<section data-v-662e624d="">
  <h3 data-v-662e624d="">Counter: 5</h3>
  <h3 data-v-662e624d="" data-testid="square-label">Square: 25</h3>
  <div data-v-662e624d=""><button data-v-662e624d="" class="btn">+1</button><button data-v-662e624d="" class="btn">-1</button></div>
</section>"
`;
